File: 1-Observer/c++/src/subject.cpp
#include <subject.h>
#include <observer.h>
#include <cstdlib>
#include <ctime>

void Subject::subscribe(Observer* newObserver) {
    _observers.push_back(newObserver);
}

void Subject::update() {
    srand(time(NULL));
    int num = rand() % 100;
    for (auto it = _observers.begin(); it != _observers.end(); it++) {
        (*it)->notify(num);
    }
}

File: 1-Observer/Java/Subject.java
import java.util.List;
import java.util.ArrayList;

public class Subject 
{
    private List<IObserver> _observers = new ArrayList<IObserver>();

    public void Subscribe(IObserver obs) 
    {
        _observers.add(obs);
    }

    public void Unsubscribe(IObserver obs)
    {
        _observers.remove(obs);
    }

    public void Updadate()
    {
        for (IObserver obs : _observers) {
            obs.Notify();
        }
    }
}